ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -630,10 +637,30 @@ func SortedGoString(v interface{}) (string, error) {
 	return string(b), nil
 }
 
-func areContainersAllEssential(ecsContainerDefs ContainerDefs) bool {
-	// NOTE: This is a silly hack for now, since the only field in a
-	// ContainerDef is essential. If/when there are other fields, we would
-	// want to change the logic for determining whether the essential field
-	// in all Containers was specified.
-	return len(ecsContainerDefs) == 0
+func hasEssential(ecsParamsContainerDefs ContainerDefs, count int) bool {
+	// If the customer does not set the "essential" field on any container
+	// definition, ECS will mark all containers in a TaskDefinition as
+	// essential. Previously, since the customer could not pass in the
+	// essential field, Task Definitions created through the CLI marked all
+	// containers as essential.
+
+	// Now that the essential field can  be set by the customer via the
+	// the ecs-params.yml file, we want to make sure that there is still at
+	// least one essential container, i.e. that the customer does not
+	// explicitly set all containers to be non-essential.
+
+	nonEssentialCount := 0
+
+	for _, containerDef := range ecsParamsContainerDefs {
+		if !containerDef.Essential {
+			nonEssentialCount += 1
+		}
+	}
+
+	// 'count' is the total number of containers specified in the service config
+	if nonEssentialCount == count {
```

Comment:
If you want, you could change these ending lines to 

`return nonEssentialCount != count`

Reply:
oh duh. good catch :)